packages/replay/src/replay.ts

```diff
@@ -392,11 +392,14 @@ export class ReplayContainer implements ReplayContainerInterface {
     const oldSessionId = this.getSessionId();
 
     // Prevent starting a new session if the last user activity is older than
-    // MAX_SESSION_LIFE. Otherwise non-user activity can trigger a new
+    // SESSION_IDLE_DURATION. Otherwise non-user activity can trigger a new
     // session+recording. This creates noisy replays that do not have much
     // content in them.
-    if (this._lastActivity && isExpired(this._lastActivity, this.timeouts.maxSessionLife)) {
-      // Pause recording
+    if (this._lastActivity && isExpired(this._lastActivity, this.timeouts.sessionIdle) && this.session?.sampled === 'session') {
```

Comment:
Updated to only pause for `session` based replays. Error based replays that idle should immediately end (as any action will start a new session which isn't what we want).

Reply:
Hmm, so what would happen to an error session in that case? If my session is error-sampled, I leave the tab for an hour and come back, recording should stop?

Would it not work to pause/resume an error session? 🤔

Reply:
@mydea In that case the session completely stops, the only way is to get resampled (e.g. reload SDK).

For all sessions, pause is effectively ending the current replay/session, but for session sampled, we want to continue listening via core SDK for a user interaction to start a new session.

For error sessions we want this to end so they get resampled otherwise it leads to a new session with 0 errors. (I suppose we could restart as error sampled session).